python/pyspark/sql/tests.py

```diff
@@ -2274,6 +2274,21 @@ def count_bucketed_cols(names, table="pyspark_bucket"):
             .mode("overwrite").saveAsTable("pyspark_bucket"))
         self.assertSetEqual(set(data), set(self.spark.table("pyspark_bucket").collect()))
 
+    def test_to_pandas(self):
+        import numpy as np
+        schema = StructType().add("a", IntegerType()).add("b", StringType())\
+                             .add("c", BooleanType()).add("d", FloatType())
+        data = [
+            (1, "foo", True, 3.0), (2, "foo", True, 5.0),
+            (3, "bar", False, -1.0), (4, "bar", False, 6.0),
+        ]
+        df = self.spark.createDataFrame(data, schema)
+        types = df.toPandas().dtypes
```

Comment:
Could we check and skip if `pandas` is not able to import (`numpy` is a Pandas dependency. So checking Pandas alone should be fine)?

```python
try:
    import pandas
    _have_pandas = True
except:
    # No Pandas, but that's okay, we'll skip those tests
    pass
...

    @unittest.skipIf(not _have_pandas, "Pandas not installed")
    def test_to_pandas(self):
        ...
```

I at least see the doctest is being skipped, `>>> df.toPandas()  # doctest: +SKIP`.